### PR TITLE
Upgrade wdl4s to 0.15 + add HTTP import resolver [GAWB-2641]

### PR DESCRIPTION
**Pull request by aednichols**
Requires https://github.com/broadinstitute/agora/pull/222 and a Cromwell upgrade

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check that the **Product Owner** has signed off on any user-facing changes
- [ ] **Submitter**: Make sure Swagger is updated if API changes
  - [ ] **...and Orchestration's Swagger too!**
- [ ] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Database checks:
  * If PR includes new or changed db queries, include the explain plans in the description
  * Make sure liquibase is updated if appropriate
  * If doing a migration, take a backup of the
  [dev](https://console.cloud.google.com/sql/instances/terraform-qfarbdq3lrexxck5htofjs5z6m/backups?project=broad-dsde-dev&organizationId=548622027621)
  and
  [alpha](https://console.cloud.google.com/sql/instances/terraform-r4caezzc35c4tb7pgdhwkmme4y/backups?project=broad-dsde-alpha&organizationId=548622027621)
  DBs in Google Cloud Console
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] Tell your tech lead (TL) that the PR exists if they want to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: core/src/test/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/SubmissionComponentSpec.scala [hunk 1]
```diff
@@ -317,15 +317,15 @@ class SubmissionComponentSpec extends TestDriverComponentWithFlatSpecAndMatchers
...
+    val result: Map[String, Map[String, Int]] = runAndWait(workflowQuery.countWorkflowsByQueueStatusByUser)
 //    result should equal(Set())
     result should contain key (testUserId)
     testUserStatusCounts.foreach { case (st, count) =>
       result(testUserId)(st.toString) should be (count)
     }
 
     // Validate all workspace counts by status
-    validateCountsByQueueStatus(result.combineAll)
+    validateCountsByQueueStatus(Monoid.combineAll(result.values))
```
Comment (aednichols) [inline core/src/test/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/SubmissionComponentSpec.scala:328]: Both [rawls](https://github.com/broadinstitute/rawls/blob/develop/project/Dependencies.scala#L58) and [wdl4s](https://github.com/broadinstitute/wdl4s/pull/255/files#diff-0ecdbc5a001d52fb34f5eafb7cd1aaa6R4) directly specify cats 0.9.0, but cats 1.0.0-MF shows up and evicts a bunch (but not all) of the sub-libraries of 0.9.0, seemingly more in `an-GAWB-2641` than in `develop`.

Unfortunately I don't have a good explanation for why cats 1.0.0-MF shows up, but its presence potentially explains the need to make this code change - there are a number of [breaking changes](https://github.com/typelevel/cats/blob/master/CHANGES.md#to-migrate-from-090) when upgrading from 0.9.0

Comment (davidangb) [reply core/src/test/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/SubmissionComponentSpec.scala:328]: wdl4s 0.15-hash depends on `async-http-client-backend-cats:0.0.16`, which depends on `cats-effect_2.11:0.4`, which depends on `cats-core_2.11:1.0.0-MF` and `cats-laws_2.11:1.0.0-MF`

* https://github.com/broadinstitute/wdl4s/blob/0.15_hotfix/project/Dependencies.scala#L39
* https://search.maven.org/remotecontent?filepath=com/softwaremill/sttp/async-http-client-backend-cats_2.11/0.0.16/async-http-client-backend-cats_2.11-0.0.16.pom
* https://search.maven.org/remotecontent?filepath=org/typelevel/cats-effect_2.11/0.4/cats-effect_2.11-0.4.pom

Comment (aednichols) [reply core/src/test/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/SubmissionComponentSpec.scala:328]: Very cool. Did you figure this out from the wdl4s `dependencyTree`? I searched through it but there were hundreds of references to cats `1.0.0-MF`, virtually all of which were just noting that `0.9.0` had been evicted

**Review by ahaessly — APPROVED** [review]
👍

File: core/src/test/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/SubmissionComponentSpec.scala (hunk 1 above)
Comment (davidangb) [reply core/src/test/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/SubmissionComponentSpec.scala:328]: yup, `sbt dependency-graph > /some/file`, then a bunch of manual hunt and peck in BBEdit.

**Review by davidangb — APPROVED** [review]
:+1: code looks good. I see tests passed in Travis; I did not spin up and test.

**Pull request merged**